From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct fib_nh_exception
Patch-mainline: never, kabi
References: kabi

In networking-stable-18_11_21, upstream commit
d52e5a7e7ca49457dd31fc8b42fb7c0d58a31221 (ipv4: lock mtu in fnhe when
received PMTU < net.ipv4.route.min_pmtu) added fnhe_mtu_locked bool to
struct fib_nh_exception. It made the kABI checker to complain.

Given the structure is internal, just hide the change from the kABI
checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/ip_fib.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/net/ip_fib.h
+++ b/include/net/ip_fib.h
@@ -57,7 +57,9 @@ struct fib_nh_exception {
 	int				fnhe_genid;
 	__be32				fnhe_daddr;
 	u32				fnhe_pmtu;
+#ifndef __GENKSYMS__
 	bool				fnhe_mtu_locked;
+#endif
 	__be32				fnhe_gw;
 	unsigned long			fnhe_expires;
 	struct rtable __rcu		*fnhe_rth_input;
